packages/flutter/test/animation/animation_controller_test.dart

```diff
@@ -769,6 +769,8 @@ void main() {
       final AnimationController controller = AnimationController(
         duration: const Duration(milliseconds: 100),
         value: 0.0,
+        lowerBound: -1,
+        upperBound: 3,
```

Comment:
Instead of changing an existing test case, could you add a new test case for this, please?

Reply:
Thanks for taking a look at the PR @goderbauer.

This existing test is supposed to test that min and max work as they should. However, it ends up only testing some _very_ special cases and therefore doesn't really work as intended. I therefore thought it would be better to add more cases to the existing test, instead of adding a new test and leaving the insufficient test unchanged.

That being said if you still prefer a new test then I'll update the PR with that 👍

Reply:
We'd still want a test that tests `.repeat` when no lowerBound and upperBound is set (basically the old test). Therefore, I'd prefer a new test case for when lowerBound and upperBound are set.

Reply:
There is [another test](https://github.com/flutter/flutter/blob/535259ce132858c17c995f8444b3bf63cb01f1ac/packages/flutter/test/animation/animation_controller_test.dart#L716) that tests `repeat` when `lowerBound` and `upperBound` is not set.

In this test, setting `lowerBound` and `upperBound` do not make a difference. The values are only used when `min` and `max` are not given to `repeat` and they are given in all the `repeat` invocations in this test. I've only set `lowerBound` and `upperBound` since there is an assert that `max <= upperBound && min >= lowerBound` which would be violated otherwise.

The changed test tests strictly more than it did before. Please consider if you don't think that's better practice than leaving a broken test in place. If not I'll create a new separate test.

Reply:
IMO, the test isn't broken. Doing what the test did before should still work after this change: you should still be able to call `repeat` with a `max`/`min` while the `AnimationController` has no `lowerBound`/`upperBound` configured. As far as I can tell, we lose test coverage for that case if we modify the test to give the `AnimationController` explicit bounds.

Reply:
> I would say that the test is broken since it doesn't test what it claims to test.

My suggestion: Let's update the name of this test then so it matches what it really tests and leave it otherwise unchanged. Then, let's add a new test case for the behavior we want to test.

Reply:
Sounds good. I'll update the PR :)